sktime/transformers/summarise.py

```diff
@@ -215,3 +228,286 @@ def transform(self, X, y=None):
+            test_for_real_target_real_feature=self.test_for_real_target_real_feature,
+            fdr_level=self.fdr_level,
+            hypotheses_independent=self.hypotheses_independent,
+            n_jobs=self.n_jobs,
+            chunksize=self.chunksize,
+            ml_task=self.ml_task
+        )
+
+        # Check to include old features
+        X_old = X
+
+        #Transform timseries data
+        X_transformed = self.feature_extractor.transform(X_old)
+        print(X_transformed.head())
+
+        self.col_to_max, self.col_to_min, self.col_to_median = get_range_values_per_column(X_transformed)
+
+        #Immpute timeseries data
+        X_transformed = impute_dataframe_range(X_transformed, col_to_max=self.col_to_max, col_to_median=self.col_to_median,
+                                             col_to_min=self.col_to_min)
+
+        # print("Imputed X is",X_transformed.head())
+        # Fit using transformed timeseries data and y
+        self.feature_selector.fit(X_transformed, y)
+
+        return self
+
+
+    def transform(self,X):
+        if self.feature_selector is None:
+            raise RuntimeError("You have to call fit before calling transform.")
+
+        if self.feature_selector.relevant_features is None:
+            raise RuntimeError("You have to call fit before calling transform.")
+
+        relevant_time_series_features = set(self.feature_selector.relevant_features) - set(pd.DataFrame(X).columns)
+        relevant_extraction_settings = from_columns(relevant_time_series_features)
+
+        # Set imputing strategy
+        #TODO keeo the impute strategy defaults.IMPUTE_FUNCTION or set it customised?
+        impute_function = partial(impute_dataframe_range, col_to_max=self.col_to_max,
+                                  col_to_min=self.col_to_min, col_to_median=self.col_to_median)
+        relevant_feature_extractor = TsFreshFeatureExtractor(kind_to_fc_parameters=relevant_extraction_settings,
+                                                      chunksize=self.feature_extractor.chunksize,
+                                                      n_jobs=self.feature_extractor.n_jobs,
+                                                      show_warnings=self.feature_extractor.show_warnings,
+                                                      disable_progressbar=self.feature_extractor.disable_progressbar,
+                                                      impute_function=impute_function,
+                                                      profile=self.feature_extractor.profile,
+                                                      profiling_filename=self.feature_extractor.profiling_filename,
+                                                      profiling_sorting=self.feature_extractor.profiling_sorting)
+        
+        Xt = relevant_feature_extractor.transform(X)
+
+        return Xt 
+
+
+    def fit_transform(self, X, y):
+        if y is None:
```

Comment:
Please use the `validate_X_y` function, should be in the same file as the `validate_X` function